bin/fetch_coreboot_crossgcc_archive.sh

```diff
@@ -0,0 +1,130 @@
+		exit 1
+	fi
+}
+
+# Delete prefix and suffix from a value
+delete_prefix_suffix() {
+	local value prefix suffix
+	value="$1"
+	prefix="$2"
+	suffix="$3"
+	value="${value/#$prefix/}"
+	value="${value/%$suffix/}"
+	echo "$value"
+}
+
+# Find the checksum file for this package
+
+# 'iasl' is special-cased.  Before coreboot 4.21, the archive was named
+# 'acpica-unix2-<ver>', and the original sources for those archives are gone.
+# Since coreboot 4.21, the archive is just named 'R<ver>.tar.gz', it lacks the
+# package name.
+# If we're fetching iasl, and this is an older release, look for the acpica
+# archive.
+if [ "$PKG_NAME" = iasl ] && [ -f "$COREBOOT_DIR/util/crossgcc/sum/"acpica-*.cksum ]; then
+	PKG_NAME=acpica
+fi
+# Otherwise, keep 'iasl' to look for the newer archive.
+
+# 'iasl' (4.21+) doesn't include the package name in the archive name, the
+# archive is just the release name
+if [ "$PKG_NAME" = "iasl" ]; then
+	PKG_CKSUM_FILE="$(single "$COREBOOT_DIR/util/crossgcc/sum/"R*.cksum)"
+else
+	PKG_CKSUM_FILE="$(single "$COREBOOT_DIR/util/crossgcc/sum/$PKG_NAME-"*.cksum)"
+fi
+
+PKG_BASENAME="$(basename "$PKG_CKSUM_FILE" .cksum)"
+# Get the base URL for the package.  This _is_ duplicated from coreboot's
+# buildgcc script, but these don't change much, and when they do we usually want
+# to use the newer source anyway for older versions of coreboot (e.g. Intel
+# broke all the iasl links - coreboot 90753398).
+case "$PKG_NAME" in
+	gmp)
+		PKG_BASEURL="https://ftpmirror.gnu.org/gmp/"
+		;;
+	mpfr)
+		PKG_BASEURL="https://ftpmirror.gnu.org/mpfr/"
+		;;
+	mpc)
+		PKG_BASEURL="https://ftpmirror.gnu.org/mpc/"
+		;;
+	gcc)
+		PKG_BASEURL="https://ftpmirror.gnu.org/gcc/gcc-$(delete_prefix_suffix "$PKG_BASENAME" gcc- .tar.xz)/"
+		;;
+	binutils)
+		PKG_BASEURL="https://ftpmirror.gnu.org/binutils/"
+		;;
+	nasm)
+		PKG_BASEURL="https://www.nasm.us/pub/nasm/releasebuilds/$(delete_prefix_suffix "$PKG_BASENAME" nasm- .tar.bz2)/"
+		;;
+	iasl)
+		PKG_BASEURL="https://github.com/acpica/acpica/archive/refs/tags/"
+		;;
+	acpica)
+		# Original acpica sources are gone.  Most of the older releases
+		# can be found here
+		PKG_BASEURL="https://distfiles.macports.org/acpica/"
+		# Version 20220331 (currently used by talos_2) isn't there, but
+		# there is an old link from Intel that is still up.  This is
+		# specific to this release.
+		if [ "$PKG_BASENAME" = acpica-unix2-20220331.tar.gz ]; then
+			PKG_BASEURL="https://downloadmirror.intel.com/774879/"
+		fi
+		;;
+esac
+
+PKG_DIGEST="$(cut -d' ' -f1 "$PKG_CKSUM_FILE")"
+
+BIN_DIR="$(dirname "${BASH_SOURCE[0]}")"
+
+# Download to packages/<arch>
+"$BIN_DIR/fetch_source_archive.sh" "$PKG_BASEURL$PKG_BASENAME" \
+	"$PKGS_DIR/coreboot-crossgcc-$PKG_BASENAME" "$PKG_DIGEST"
+
+# Copy to the tarballs directory so coreboot's toolchain build will use this
+# archive
+mkdir -p "$COREBOOT_DIR/util/crossgcc/tarballs"
+cp "$PKGS_DIR/coreboot-crossgcc-$PKG_BASENAME" \
```

Comment:
@JonathonHall-Purism `ln -s` ? `cp -a` ?

Reply:
Yeah a symlink should be fine here, might save me a few GB of disk space given all the PB/Heads workspaces I have :grin:   I'll change it

Reply:
@JonathonHall-Purism cp -l might be better (hardlink)

Reply:
(Discussed out of band, agreed on symlink)

Reply:
Changed to `ln -s` in https://github.com/linuxboot/heads/pull/1576/commits/93ccf25d24bb73c89ff51e17d847bccc78bc61e2